android/app/src/main/java/com/example/viewport/ui/ViewportScreens.kt

```
@@ -0,0 +1,943 @@
+
+@Composable
+private fun ShareLinkComposer(label: String, password: String, onLabel: (String) -> Unit, onPassword: (String) -> Unit, onCreate: () -> Unit) {
+  SectionCard(title = "Create share link", icon = Icons.Outlined.Link) {
+    OutlinedTextField(label, onLabel, label = { Text("Label") }, modifier = Modifier.fillMaxWidth(), singleLine = true, shape = MediumShape)
+    OutlinedTextField(password, onPassword, label = { Text("Password optional, 8-72 bytes") }, modifier = Modifier.fillMaxWidth(), singleLine = true, visualTransformation = PasswordVisualTransformation(), shape = MediumShape)
+    Button(onClick = onCreate, modifier = Modifier.fillMaxWidth(), shape = MediumShape) { Text("Create link") }
+  }
+}
+
+@Composable
+private fun ShareLinkCard(link: ShareLinkResponse, onToggle: (() -> Unit)? = null, onDelete: (() -> Unit)? = null) {
```

detekt (1.23.8)

[warning] The function ShareLinkCard appears to be too complex based on Cognitive Complexity (complexity: 30). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)